### PR TITLE
Revise help text for pool metadata source CLI option

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-482

### Overview

Noticed while reviewing input-output-hk/cardano-launcher#90.

- Fix variable names to reflect the new option
- Add more detail to CLI help text

### CODE REVIEW

**Review by hasufell — APPROVED** [review]

File: lib/cli/src/Cardano/CLI.hs [hunk 1]
```diff
@@ -1318,15 +1318,16 @@ tlsOption = TlsConfiguration
...
-    <> help "Pool Metadata fetching strategy. This setting will persist across restarts. Possible values: \
-            \ none, \
-            \ direct, \
-            \ <SMASH-URL>."
+    <> metavar "( none | direct | SMASH-URL )"
+    <> help ("Stake pool metadata fetching strategy. "
+            <> "Provide a URL to specify a SMASH metadata proxy server, "
+            <> "use \"direct\" to fetch directly from the registered pool URLs,"
+            <> " or \"none\" to completely disable stake pool metadata. "
+            <> "This setting will persist across restarts.")
```
Comment (KtorZ) [inline lib/cli/src/Cardano/CLI.hs]: Perhaps: "This setting will persist across restarts AND override any previous setting set via this option or via the API." ?

**Review by KtorZ — APPROVED** [review]

File: lib/cli/src/Cardano/CLI.hs (hunk 1 above)
Comment (hasufell) [reply lib/cli/src/Cardano/CLI.hs]: I'm starting to think it was a bad idea to keep the cli switch. It's confusing. Now someone might still think it **pins** the setting or something.

Comment (rvl) [reply lib/cli/src/Cardano/CLI.hs]: I think it's fine to have a sticky setting like this. Makes sense to me. Anyway, our users are likely to _either_ configure this by the CLI, _or_ by API, but not both.

Reworded a bit more.

**Review by KtorZ — APPROVED** [review]

**Comment by KtorZ** [comment]
bors merge

**Pull request merged**